db/blob/blob_file_builder.cc

```diff
@@ -253,19 +267,18 @@ Status BlobFileBuilder::CompressBlobIfNeeded(
   constexpr uint32_t compression_format_version = 2;
 
   bool success = false;
-
+  std::string output;
   {
     StopWatch stop_watch(immutable_options_->clock, immutable_options_->stats,
                          BLOB_DB_COMPRESSION_MICROS);
-    success =
-        CompressData(*blob, info, compression_format_version, compressed_blob);
+    success = CompressData(blob, info, compression_format_version, &output);
   }
 
   if (!success) {
     return Status::Corruption("Error compressing blob");
   }
 
-  *blob = Slice(*compressed_blob);
+  *compressed_blob = Slice(output);
```

Comment:
After this function returns, output is destroyed, but `compressed_blob` still points to `output.data()`.

Reply:
Then, I'll pass in std::string* and create a slice in the upper function.